plugins/modules/system/sap_task_list_execute.py

```diff
@@ -0,0 +1,366 @@
+    host: 10.1.8.10
+    sysnr: '00'
+    client: '000'
+    task_to_execute: SAP_BASIS_SSL_CHECK
+    task_parameters :
+      - { 'TASKNAME': 'CL_STCT_CHECK_SEC_CRYPTO', 'FIELDNAME': 'P_OPT2', 'VALUE': 'X' }
+      - TASKNAME: CL_STCT_CHECK_SEC_CRYPTO
+        FIELDNAME: P_OPT3
+        VALUE: X
+    task_settings: batch
+
+# Exported environement variables.
+- name: Hint if module will fail with error message like ImportError libsapnwrfc.so...
+  community.general.sap_task_list_execute:
+    conn_username: DDIC
+    conn_password: Passwd1234
+    host: 10.1.8.10
+    sysnr: '00'
+    client: '000'
+    task_to_execute: SAP_BASIS_SSL_CHECK
+    task_settings: batch
+  environment:
+    SAPNWRFC_HOME: /usr/local/sap/nwrfcsdk
+    LD_LIBRARY_PATH: /usr/local/sap/nwrfcsdk/lib
+'''
+
+RETURN = r'''
+msg:
+  description: A small execution description
+  type: str
+  returned: always
+  sample: 'Successfull'
+out:
+  description: A complete description of the executed tasks. If this is available.
+  type: list
+  elements: dict
+  returned: on success
+  sample: [...,{
+              "LOG": {
+                  "STCTM_S_LOG": [
+                      {
+                          "ACTIVITY": "U_CONFIG",
+                          "ACTIVITY_DESCR": "Configuration changed",
+                          "DETAILS": null,
+                          "EXEC_ID": "20210728184903.815739",
+                          "FIELD": null,
+                          "ID": "STC_TASK",
+                          "LOG_MSG_NO": "000000",
+                          "LOG_NO": null,
+                          "MESSAGE": "For radiobutton group ICM too many options are set; choose only one option",
+                          "MESSAGE_V1": "ICM",
+                          "MESSAGE_V2": null,
+                          "MESSAGE_V3": null,
+                          "MESSAGE_V4": null,
+                          "NUMBER": "048",
+                          "PARAMETER": null,
+                          "PERIOD": "M",
+                          "PERIOD_DESCR": "Maintenance",
+                          "ROW": "0",
+                          "SRC_LINE": "170",
+                          "SRC_OBJECT": "CL_STCTM_REPORT_UI            IF_STCTM_UI_TASK~SET_PARAMETERS",
+                          "SYSTEM": null,
+                          "TIMESTMP": "20210728184903",
+                          "TSTPNM": "DDIC",
+                          "TYPE": "E"
+                      },...
+                    ]}}]
+'''
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils.json_utils import json
+from xml.etree.ElementTree import XML
+import traceback
+try:
+    from pyrfc import Connection
+except ImportError:
+    HAS_PYRFC_LIBRARY = False
+    PYRFC_LIBRARY_IMPORT_ERROR = traceback.format_exc()
+else:
+    HAS_PYRFC_LIBRARY = True
+
+
+def xml_dict(xml_raw):
+    # adapted from: http://code.activestate.com/recipes/410469-xml-as-dictionary/
```

Comment:
Hmm, what's with the copyright and license for this snippet? It's very close to the original at https://code.activestate.com/recipes/410469-xml-as-dictionary/, so I think we need the author's approval that this can be licensed as GPLv3+. Right now, it is licensed under the [PSF license](https://en.wikipedia.org/wiki/Python_Software_Foundation_License) according to https://code.activestate.com/recipes/410469-xml-as-dictionary/.

Reply:
Hi, the answer was no so i changed it to the dicttoxml module

Reply:
Ah, that's a good solution :)